### ISSUE TITLE
strikeout formatting on bulleted list item garbles markup

### ISSUE THREAD

**Issue by gmorten1** [tomboy-notes/tomboy-ng#182]
if you select some text inside a bulleted list and apply the strikeout formatting (ctrl-s), the strikeout tag overlaps with the list-item container and causes errors on opening and indexing, such as:

Unmatching element end tag (expected "</list-item>")

**Comment by gmorten1** [comment]
the behavior described above doesn't happen in all cases, but i can duplicate on demand by selecting text from the beginning of one bullet to the middle of the next bullet and applying the strikeout style. this results in bad xml in the .note file:

```
testing
<list><list-item dir="ltr"><strikeout>line 1 is here</strikeout></list-item></list><strikeout></strikeout>
<list><list-item dir="ltr">and line 2</strikeout> is here</list-item></list><strikeout>
<list><list-item dir="ltr">and line 3 is here</list-item></list>
```
note the unmatched strikeout end tag in the middle of line 2 and unclosed strikeout tag after line 2. 

as far as i can tell, this never happens with the bold style, which correctly places bold tags inside the list-items. it also sometimes places an empty bold tag pair `<bold></bold>` outside the list container but that seems harmless (?). that also happens with the strikeout style (see line 1 in the example xml above)

**Comment by davidbannon** [comment]
Ah, thanks for identifying that gmorten1, I have to say that the bullet code is the part of tomboy-ng I am least happy with !  It overrides the code thats built into KMemo thats even worse !

I am not particularly surprised there is a still a problem in there, I'll dig into it again later today if I get a chance, thanks for the report ....
Davo

**Comment by davidbannon** [comment]
Yep, found it.  If you want me to cut you a fixed package, please let me know your OS.
That bug has been there since strikeout was added. Sigh ....
Thanks for the heads up !

Davo

**Comment by gmorten1** [comment]
thanks for the quick fix!

i'm using a .deb from ubuntu repo (i current have .28b) but i can wait until it shows up in the repo unless that's going to take forever(ish)

really appreciate the work you're doing on this. it's great to be able to use tomboy without the mono bloat and resource piggishness. i've been using tomboy for years and have thousands of notes that i sync through dropbox. now, about those multilevel bullets.... ;)

**Comment by davidbannon** [comment]
> multilevel bullets

Hmm, not a term I am familiar with.  
Seriously, I don't think its going to happen while I am using KMemo (it provides the editing component), I am watching closely the development of a rival component, RichMemo but its got a long way to go .....
I guess I could hurry it up a bit.
Davo

**Comment by davidbannon** [comment]
Fixed in v0.29

Thanks for the report.

Davo
